### PR TITLE
Remove duplication in hyperloglog code

### PR DESCRIPTION
**Pull request by oarap**
We had significant duplication in hyperloglog extension and utility
library that we use in the analyze related code. This PR removes the
duplication as well as significant amount dead code. It also fixes
some compiler warnings and some coverity issues.

This PR also puts the hyperloglog functions in a separate schema
which is non-modifiable by non superusers.

### CODE REVIEW

**Comment by danielgustafsson** [comment]
Since this is quite a large commit, am I parsing it right if I think you are removing the contrib module completely and prune all the unused HLL functionality from the backend copy? Since we are refactoring the code anyways, is there a reason to keep multiple headers and creating a hierarchy or can we amalgamate to just providing `src/backend/utils/hyperloglog.h`?

Regarding the commit message, since you are effectively deduplicating the code the sentence "Remove deduplication" reads wrong. It should either "Deduplication .." or "Remove duplication ..".

**Comment by oarap** [comment]
The contrib module stays for now but it is using the code from the utility where we migrated and updated hyperloglog code. If we decide and achieve to put the hyperloglog functions into catalog, it will also go away.  

Creating one single header `hyperloglog.h` instead of having `constants.h   encoding.h    hllutils.h    hyperloglog.h` is a great idea. I will do it today and update the PR. Thanks for the feedback.

File: src/backend/commands/analyze.c [hunk 1]
```diff
@@ -3384,7 +3383,7 @@ compute_scalar_stats(VacAttrStatsP stats,
 		{
 			HLLCounter hLLFull = (HLLCounter) DatumGetByteaP(stats->stahll_full);
 			HLLCounter hllFull_copy = hll_copy(hLLFull);
-			stats->stadistinct = hyperloglog_get_estimate(hllFull_copy);
+			stats->stadistinct = round(hyperloglog_estimate(hllFull_copy));
```
Comment (khannaekta) [inline src/backend/commands/analyze.c]: Why do we roundoff the hyperloglog estimated distinct value ?

Comment (oarap) [reply src/backend/commands/analyze.c]: In some cases, where there is one single value, NDV might be reported as less than 1 such as 0.9999. I didn't want to create any confusion as fractions have a different meaning in gpdb (postgres) even though we convert them into negative number.

**Pull request merged**